src/tiledimage.js

```diff
@@ -967,6 +1017,45 @@ $.extend($.TiledImage.prototype, $.EventSource.prototype, /** @lends OpenSeadrag
         this._raiseBoundsChange();
     },
 
+    /**
+     * Get the region of this tiled image that falls within the viewport.
+     * @returns {OpenSeadragon.Rect} the region of this tiled image that falls within the viewport.
+     */
+    getDrawArea: function(){
+
+        var drawArea = this._viewportToTiledImageRectangle(
+            this.viewport.getBoundsWithMargins(true));
+
+        if (!this.wrapHorizontal && !this.wrapVertical) {
+            var tiledImageBounds = this._viewportToTiledImageRectangle(
+                this.getClippedBounds(true));
+            drawArea = drawArea.intersection(tiledImageBounds);
+        }
+
+        return drawArea;
+    },
+
+    /**
+     *
+     * @returns {Array} Array of Tiles that make up the current view
+     */
+    getTilesToDraw: function(){
+
+        let tileArray = this._tilesToDraw.flat();
+        // update all tiles (so blending can happen right at the time of drawing)
+        this._updateTilesInViewport(tileArray);
```

Comment:
Can you explain the purpose of this code? Why isn't `getTilesToDraw` just returning the flattened tile array directly?

Reply:
The operations that trigger `TiledImage.update()` and `World.draw()` aren't as tightly coupled anymore. Immediately before each draw operation, the tiles need a chance to update their opacity to account for blending animations. As they become opaque, the coverage of lower levels can change, and some tiles will not need to be rendered any longer, nor have their `beingDrawn` property set to `true` (which affects the cache).

I've expanded the comment a bit too.

Reply:
Great, thank you. Good to have these things documented!